pandas/core/arrays/datetimes.py

```diff
@@ -504,8 +505,6 @@ def _assert_tzawareness_compat(self, other):
 
     def _sub_datetime_arraylike(self, other):
         """subtract DatetimeArray/Index or ndarray[datetime64]"""
-        if len(self) != len(other):
-            raise ValueError("cannot add indices of unequal length")
```

Comment:
This may have been removed by mistake, since the decorator isn't yet enabled in this module